portfolio/src/main/java/com/google/sps/servlets/DisplayBlobServlet.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This servlet is called to display images stored in Blobstore given their blobKey
+ */
+@WebServlet("/display-blobstore")
+public class DisplayBlobServlet extends HttpServlet {
+
+  @Override
+  /** 
+    * Handles GET requests to display the image stored in Blobstore. The serve function arranges
+    * for blob with the blobKey specified in the qury string to be served as the response content
+    * for the current request. 
+    */
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    BlobKey blobKey = new BlobKey(request.getParameter("blob"));
+
+    blobstoreService.serve(blobKey, response);
```

Comment:
What happens if there was no `blob` parameter, or if the blob key was not found in blobstore?

Might be worth putting in some sanity checks so that you can gracefully handle bad requests (maybe blobstore handles the 404 case?, but we should certainly make sure that the request parameter is there)

Reply:
+1

Reply:
I changed the code to send a HTTP 400 Bad Request response. 
The case when `blob `parameter is null is considered in script.js, however it is possible that the key was not found in blobstore.